### PR TITLE
feat: Add opencensus client

### PR DESCRIPTION
**Pull request by frekw**
Hi!

This adds an [OpenCensus](https://opencensus.io/) client to `zio-telemetry`. The reason I needed it was because it has great integration with [Google Cloud Trace/Stackdriver Tracing](https://cloud.google.com/trace).

It largely follows the structure of the existing clients. The difference is that the `Service` does a bit more than the current clients. This is due to the fact that I've extracted it out of an application, and there it was really useful to have an injectable service that was a bit more capable for adding tracing to other services such as a Redis client wrapper, without having the change the interfaces from e.g `get[A](key: String): ZIO[Any, Throwable, A]` to `get[A](key: String): ZIO[Tracing, Throwable, A]`.

It currently lacks tests because I haven't been able to find a good mocked or inmemory exporter that could be used for testing (looking at `opencensus-java`, it doesn't seem like just mocking the `Tracer` itself will give much).

I've chosen to hold off adding any examples awaiting https://github.com/zio/zio-telemetry/issues/106, but could quite easily extract an example on top of e.g [zio-grpc](https://github.com/scalapb/zio-grpc) which I'm currently using it with. I'm not super familiar with `http4s` but I assume adding an example for it would be pretty straight forward as well.

### CODE REVIEW

File: opencensus/src/main/scala/zio/telemetry/opencensus/Live.scala [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+  def root[R, E, A](
+    name: String,
+    kind: Span.Kind,
+    toErrorStatus: ErrorMapper[E],
+    attributes: Map[String, AttributeValue]
+  )(effect: ZIO[R, E, A]): ZIO[R, E, A] =
+    for {
+      res <- createSpan(BlankSpan.INSTANCE, name, kind).use(
+              finalizeSpanUsingEffect(
```
[the remark this answers is not in the document]
Comment (frekw) [reply opencensus/src/main/scala/zio/telemetry/opencensus/Live.scala:52]: Sure! What did you have in mind, something like this?

```scala
createSpanFromRemote(remote, name, kind).use(
  finalizeSpanUsingEffect(putAttributes(attributes) *> effect, toErrorStatus)(_)
)
```

?

**Comment by frekw** [comment]
Thanks so much for the review @mijicd. I've addressed most comments.

I think I'll try to put together an example once https://github.com/scalapb/zio-grpc/issues/59 is resolved since I think it would benefit both projects! Maybe it makes sense to put together some basic usage in the README in either case?

File: opencensus/src/main/scala/zio/telemetry/opencensus/Live.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (mijicd) [reply opencensus/src/main/scala/zio/telemetry/opencensus/Live.scala:52]: Yes, you can even drop the `(_)` part.

**Comment by mijicd** [comment]
@frekw that would be awesome!

File: opencensus/src/main/scala/zio/telemetry/opencensus/Live.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (frekw) [reply opencensus/src/main/scala/zio/telemetry/opencensus/Live.scala:51]: 👍

**Comment by runtologist** [comment]
> Looks good, a couple of remarks from my side.It would be great if you could provide a small usage example in readme.
> 
> @runtologist all yours :)

Thanks. I'm quite busy atm. I trust you guys to make this awesome! ;-)

**Comment by frekw** [comment]
Nice, thanks @mijicd.

I'll try to put together an end to end example sometime this week.

I think one thing that we'll need to discuss in that PR is a good API is for extracting/injecting trace information when accessing/modifying headers is effectful. zio-grpc recently [changed metadata interactions to be effectful](https://github.com/scalapb/zio-grpc/commit/7945c5dfc3cfcdaaead8d1a2d8bfb35f949527ec) since doing so may not be thread safe. That seems like it could be a common pattern and something that we need to support. However, it does make the interop with Java side of things slightly less obvious than it is today.

**Pull request merged**